ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v1/container/clientlibs/editor/js/editDialog.js

```diff
@@ -42,6 +42,259 @@
+
+    var formModelChangeConfirmationDialog = new Coral.Dialog().set({
+        id: 'formModelChange',
+        header: {
+            innerHTML: dialogHeader
+        },
+        content: {
+            innerHTML: dialogContent
+        },
+        footer: {
+            innerHTML: dialogFooter
+        },
+        variant : 'warning'
+    });
+
+    channel.on("dialog-success", function (e) {
+        configuredFormModel = toBeConfiguredFormModel;
+        isConfirmationDialogAccept = false;
+    });
+
+    channel.on('change', '#doNotShowDialogCheckBox', function(e) {
+        if (this.checked) {
+            doNotShowDialogFlag = true;
+        } else {
+            doNotShowDialogFlag = false;
+        }
+    });
+
+    function selectFormModelOnLoad(dialog) {
+        var schemaType = dialog.find(FORM_MODEL_SELECTOR)[0].value;
+        hideContainersExcept(schemaType);
+        prefillSchema(schemaType, dialog);
+        if (!isTemplate() && schemaType != NONE) {
+            document.querySelector(FORM_MODEL_SELECTOR).disabled = true;
+            dialog.find(SCHEMA_TYPE)[0].removeAttribute("disabled");
+        }
+        document.body.appendChild(formModelChangeConfirmationDialog);
+    };
+
+    function selectFormModelOnChanged(dialog) {
+        var schemaTypeSelected = dialog.find(FORM_MODEL_SELECTOR)[0].value;
+        hideContainersExcept(schemaTypeSelected);
+        if (schemaTypeSelected == SCHEMA) {
+            if (lastSelectedXsd) {
+                $(SCHEMA_ROOT_ELEMENT_SELECTOR).show();
+                createRootElementSelectionMarkup(lastSelectedXsd);
+            }
+        } else {
+            removeXsdRootElementMarkup();
+        }
+    };
+
+    function prefillSchema(schemaType, dialog){
+        var schemaRef = dialog.find(SCHEMA_REF)[0].value,
+            xsdRef = dialog.find(XSD_REF)[0].value;
+        if (schemaType == SCHEMA) {
+            $(SCHEMA_DROPDOWN_SELECTOR).val(schemaRef ? schemaRef : xsdRef);
+        } else if (schemaType == FORM_DATA_MODEL) {
+            $(FDM_DROPDOWN_SELECTOR).val(schemaRef ? schemaRef : xsdRef);
+        }
+        if(xsdRef){
+            fillXsdDetails(xsdRef);
+        }
+    };
+
+    function schemaSelectorOnChanged(dialog) {
+        var selectedSchema = dialog.find(SCHEMA_DROPDOWN_SELECTOR)[0].value;
+        if (selectedSchema.indexOf(XSD_EXTENSION) != -1) {
+            $(SCHEMA_ROOT_ELEMENT_SELECTOR).show();
+            dialog.find(XSD_REF)[0].value = selectedSchema;
+            dialog.find(SCHEMA_REF)[0].value = '';
+            createRootElementSelectionMarkup(selectedSchema);
+        } else if (selectedSchema.indexOf(JSON_EXTENSION) != -1) {
+            $(SCHEMA_ROOT_ELEMENT_SELECTOR).hide();
+            removeXsdRootElementMarkup();
+            handleJsonSelection(selectedSchema);
+            dialog.find(SCHEMA_REF)[0].value = selectedSchema;
+            dialog.find(XSD_REF)[0].value = '';
+        } else {
```

Comment:
we are not supporting XSD as form model in AF2 (yet). pls remove all code related to XSD.

Reply:
noted

Reply:
we are still listing xsd in AF2 although support is not there, will need to work on that too